### ISSUE TITLE
[Tech Debt]: Introduce Cargo Feature Flag to Safeguard Incomplete Withdrawal Functionality

### ISSUE THREAD

**Issue by Jiloc** [stacks-network/sbtc#960]
## Description

PR that introduced tech-debt: #849 

Add a Cargo feature flag to disable the processing of withdrawal events by default. While we already prevent withdrawals from being processed [during validation](https://github.com/stacks-network/sbtc/blob/main/signer/src/bitcoin/validation.rs#L701), this additional safeguard ensures the functionality remains disabled unless explicitly enabled for development purposes. 
This prevents accidental activation due to unintentional code changes.

**Comment by aldur** [comment]
> PR that introduced tech-debt: https://github.com/stacks-network/sbtc/pull/849

Did we add code for withdrawals in this PR, specifically?

**Comment by Jiloc** [comment]
Not really, it’s the PR that added validation, so I probably shouldn’t have marked it specifically. I believe the (partial) withdrawal support itself was introduced across multiple PRs alongside the deposits functionality. For example, when we added support in the block_event to capture deposits prints events from the smart contracts, we also did the same for withdrawals. If you think it’s worth investigating further, I can compile a list of related PRs

**Comment by matteojug** [comment]
Currently the withdrawal functionality (and access to the relative code paths) are prevent via:
 - There are no calls to `write_withdrawal_request`: we never (outside tests) call the function to store the withdrawal requests in the db.
 - Even with those requests in the db, they would fail bitcoin validation: we purposely fail validation for withdrawals tx
